src/report-filter-controls.tsx

```diff
@@ -1,48 +1,30 @@
-import { createContext, VNode } from "preact";
+import { VNode } from "preact";
 import { useRef, useContext } from "preact/hooks";
 import { utcFormat } from "d3-time-format";
 import { utcWeek, CountableTimeInterval } from "d3-time";
-import { route } from "./router";
 import { AgenciesContext } from "./context/agencies-context";
+import { ReportFilterContext, DEFAULT_ENV } from "./context/report-filter-context";
 
 const yearMonthDayFormat = utcFormat("%Y-%m-%d");
-const DEFAULT_IAL = 1;
-const DEFAULT_ENV = "prod";
 
-interface ReportFilterControlsContextValues {
-  start: Date;
-  finish: Date;
-  ial: 1 | 2;
-  agency?: string;
-  env: string;
-}
-
-const ReportFilterControlsContext = createContext({
-  start: new Date(),
-  finish: new Date(),
-  ial: DEFAULT_IAL,
-  env: DEFAULT_ENV,
-} as ReportFilterControlsContextValues);
-
-interface ReportFilterControlsProps {
-  path: string;
-}
-
-function ReportFilterControls({ path }: ReportFilterControlsProps): VNode {
-  const { start, finish, agency, ial, env } = useContext(ReportFilterControlsContext);
+function ReportFilterControls(): VNode {
+  const { start, finish, agency, ial, env, setParameters } = useContext(ReportFilterContext);
   const { agencies } = useContext(AgenciesContext);
 
   const formRef = useRef(null as HTMLFormElement | null);
 
-  function update(event: Event, overrideFormData = {}) {
+  function update(event: Event, overrideFormData: Record<string, string> = {}) {
     const form = formRef.current;
     if (!form) {
       return;
     }
+
     const formData = Array.from(new FormData(form)) as string[][];
-    const seachParams = new URLSearchParams(formData);
-    Object.entries(overrideFormData).forEach(([key, value]) => seachParams.set(key, String(value)));
-    route(`${path}?${seachParams.toString().replace(/\+/g, "%20")}`);
+    const searchParams = new URLSearchParams(formData);
+    Object.entries(overrideFormData).forEach(([key, value]) =>
+      searchParams.set(key, String(value))
+    );
+    setParameters(Object.fromEntries(searchParams));
```

Comment:
At a glance, it looks like it can work to just call `Object.fromEntries` on form data directly?

```suggestion
    setParameters(Object.fromEntries(new FormData(form)));
```

Reply:
Gave a shot in 9321f8bbb56ea5aeabe858da7b0ba0b4d8b14e4d